### DISCUSSION TITLE
Generating a PDF from docs/*.md

### DISCUSSION THREAD

**Discussion by gvanem** [microsoft/vcpkg#19349] [Q&A]
I've been struggling creating a single `VCPKG.PDF` from all `docs/*.md` files with little success. 
First trying with Pandoc (latest v2.14.1), but it does not follow and convert internal links like:
```
  - [FAQ](about/faq.md)
```
to a sensible link and content. This was my command:
```
pandoc  --output VCPKG.pdf --pdf-engine xelatex --metadata title="VCPKG documentation" ^
  --table-of-contents --listings --self-contained --reference-links --to=pdf --from=gfm README.md 
```
I assume some Pandoc filter is required for this. But I'm not so into this. Anybody?

Then a tip in https://github.com/microsoft/vcpkg/pull/1111/files hinted on trying:
```
pwsh docs\regenerate.ps1 -VcpkgRoot . -WarningAction Stop
```

which did absolutely nothing.

So how should I create a single offline `VCPKG.pdf` file?

**Comment by PhoebeHui** [comment]
@gvanem, I haven't used pandoc before, I have no idea why it doesn't work here.

~~I know a simple way to same the .md file to .pdf, however, I'm not sure if it matches your requirements:
Open the .md file with Chrome -> Print -> Destination: Save as PDF -> Save~~

**Reply by gvanem** [reply]
@PhoebeHui Read my Q again. I wrote **from all** `docs/*.md` files.

**Reply by PhoebeHui** [reply]
Sorry, forget my previous comments.

**Reply by gvanem** [reply]
I'd also settle for a `VCPKG.CHM` file. In Python3 it's easy as:
```
cd Doc
make htmlhelp
cd build\htmlhelp
hhc python3110a0.hhp
```
Presto; this ends up with a beautiful 10 MByte CHM file. 
You could to it equally easy.

**Comment by gvanem** [comment]
Is it possible to use AsciiDoctor for such a task? I.e. create a single .PDF/.CHM from all the `%VCPKG_ROOT%/docs/*.md` files?
Since the `regenerate.ps1` script does not work for me and is so obfuscated.
